### ISSUE TITLE
mognodb database name is hardcoded in all Models

### ISSUE THREAD

**Issue by pkjoshisbp** [yiisoft/yii2-mongodb#296]
### What steps will reproduce the problem?
Generate any model with MongoGii.

### What is the expected result?
The model should take only collection name like mysql table. else problem is, if I copy the setup to another instance, I need to manually find and replace all occurrences of database-name. when we are already providing the database name in config, why it needs the db info from model.

### Additional info

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.31
| PHP version      | 7.2.24
| Operating system | Ubuntu Linux 18.04

**Comment by samdark** [comment]
The solution sounds like the one breaking backwards compatibility. Am I correct? We may do it in 2.2.0.

**Comment by pkjoshisbp** [comment]
I didn't get what you mean by "breaking backwards compatibility".  What I am trying to say is like say I have one yii2 where I have 50 collections.
I copy the whole yii2 setup for another application for reuse. Now in the new setup the database name is different. Since db-name is included in all the 50 models, I have to replace all occurrences of db-name in all models.
In mysql model we return only table name, so I think we can use only collection-name for mongodb model.
please let me know if you need any other info.

**Comment by samdark** [comment]
"breaking backwards compatibility" means that if we'll replace database name with collection name, existing applications that relied on the old behavior may break.

**Comment by pkjoshisbp** [comment]
yes samdark. that is correct. if we change the database name only in config file and don't make change in all models it will break. collection-name remaining same.
